addons/project/models/project.py

```diff
@@ -1088,6 +1090,20 @@ def _read_group_personal_stage_type_ids(self, stages, domain, order):
     # customer portal: include comment and incoming emails in communication history
     website_message_ids = fields.One2many(domain=lambda self: [('model', '=', self._name), ('message_type', 'in', ['email', 'comment'])])
     is_private = fields.Boolean(compute='_compute_is_private')
+    allow_milestones = fields.Boolean(related='project_id.allow_milestones')
+    milestone_id = fields.Many2one(
+        'project.milestone',
+        'Milestone',
+        domain="[('project_id', '=', project_id)]",
+        compute='_compute_milestone_id',
+        readonly=False,
+        store=True,
+    )
+    has_late_and_not_reached_milestone = fields.Boolean(
```

Comment:
Maybe a bit nitpicky, but I would replace the occurrences of `/not_reached(?=_milestones?)/` with the proper word `unreached`, to make it clear that this isn't the negation of another value.

Reply:
Indeed, I agree, thanks. 🙂